torch/fx/experimental/symbolic_shapes.py

```diff
@@ -1424,6 +1433,458 @@ def rbinary_magic_impl(self, other):
+            def wrapper(*args):
+                # Lifts the arguments into a list of 'Z3Node'.
+                args = tuple(wrap(a).expr for a in args)
+                # Run the function on the Z3 expressions.
+                r = func(*args)
+                # Lifts the return value into a 'Z3Node', if it's not
+                # already an object of that type.
+                assert isinstance(r, (z3.ArithRef, z3.BoolRef, Z3Node))
+                return r if isinstance(r, Z3Node) else Z3Node(r)
+
+            return wrapper
+
+        replacement_map = {
+            # Operator module.
+            operator.not_: lift(z3.Not),
+            operator.and_: lift(z3.And),
+            operator.or_: lift(z3.Or),
+            operator.floordiv: lift(Z3Ops.floordiv),
+            operator.truediv: lift(Z3Ops.div),
+            operator.mod: lift(Z3Ops.mod),
+
+            # Math module.
+            math.ceil: lift(Z3Ops.ceil),
+            math.floor: lift(Z3Ops.floor),
+
+            # Torch module.
+            torch.sym_float: lift(Z3Ops.to_real),
+            sym_sqrt: lift(Z3Ops.sqrt),
+            # Not lifted because we only use this function as a
+            # marker for adding the expression as validator input.
+            torch._assert: torch._assert,
+        }
+        return replacement_map[op] if op in replacement_map else lift(op)
+
+    # Processes an FX graph, populating the given validator.
+    #
+    # This function walks through each node in the FX graph, translating
+    # them into the Z3 world.
+    #
+    # Then, whenever it finds an 'torch._assert' call_function operation,
+    # it adds the Z3 expression corresponding to the argument as validator
+    # input.
+    def _populate_validator_with_fx_graph(
+            validator: "TranslationValidator",
+            graph: torch.fx.Graph
+    ) -> None:
+        node_to_z3: Dict[torch.fx.Node, Z3Node] = {}
+
+        # Creates a Z3 variable, taking into account the node type.
+        def run_placeholder(node: torch.fx.Node) -> None:
+            assert "symbol" in node.meta
+            node_to_z3[node] = Z3Node(validator.z3var(node.meta["symbol"]))
+
+        # Actually runs the node target function (which is already
+        # lifted) with its arguments.
+        def run_call_function(node: torch.fx.Node) -> None:
+            args = [
+                node_to_z3[a]
+                if isinstance(a, torch.fx.Node) else a
+                for a in node.args
+            ]
+
+            assert callable(node.target)
+            node_to_z3[node] = node.target(*args)
+            validator.add_assertion(node_to_z3[node].assertion)
+
+        # Adds the Z3 expression corresponding to the first argument
+        # as a validator input.
+        def run_assertion(node: torch.fx.Node) -> None:
+            assertion = node.args[0]
+            assert isinstance(assertion, torch.fx.Node)
+            validator.add_input(node_to_z3[assertion].bool)
```

Comment:
It's a bit weird for this to be called input. Maybe?

Reply:
Agreed. I also find myself getting confused more often than not.